File: app/src/main/java/hu/bme/aut/tvshows/network/api/PeopleApi.java
package hu.bme.aut.tvshows.network.api;//retrofit2

import java.util.List;

import hu.bme.aut.tvshows.model.Person;
import hu.bme.aut.tvshows.model.PersonData;
import hu.bme.aut.tvshows.model.PersonSearchResult;
import retrofit2.Call;
import retrofit2.http.DELETE;
import retrofit2.http.GET;
import retrofit2.http.Headers;
import retrofit2.http.POST;
import retrofit2.http.PUT;

public interface PeopleApi {
  /**
   * Delete person
   * 
   * @param personId  (required)
   * @return Call&lt;Void&gt;
   */
  @DELETE("people/{personId}")
  Call<Void> deletePeoplePersonId(
            @retrofit2.http.Path("personId") Integer personId            
  );

  /**
   * Get person by ID
   * Retrieve all primary information for a given person.
   * @param personId  (required)
   * @return Call&lt;Person&gt;
   */
  @GET("people/{personId}")
  Call<Person> getPeoplePersonId(
            @retrofit2.http.Path("personId") Integer personId            
  );

  /**
   * Search people
   * 
   * @param q search text (optional)
   * @return Call&lt;List&lt;PersonSearchResult&gt;&gt;
   */
  @GET("search/people")
  Call<List<PersonSearchResult>> getSearchPeople(
        @retrofit2.http.Query("q") String q                
  );

  /**
   * Create person
   * 
   * @param body  (optional)
   * @return Call&lt;Person&gt;
   */
  @Headers({
    "Content-Type:application/json"
  })
  @POST("people")
  Call<Person> postPeople(
                    @retrofit2.http.Body PersonData body    
  );

  /**
   * Update person
   * 
   * @param personId  (required)
   * @param body  (optional)
   * @return Call&lt;Void&gt;
   */
  @Headers({
    "Content-Type:application/json"
  })
  @PUT("people/{personId}")
  Call<Void> putPeoplePersonId(
            @retrofit2.http.Path("personId") Integer personId            ,                 @retrofit2.http.Body PersonData body    
  );

}

File: app/src/main/java/hu/bme/aut/tvshows/network/api/PeopleApi.kt
package hu.bme.aut.tvshows.network.api

import hu.bme.aut.tvshows.model.Person
import hu.bme.aut.tvshows.model.PersonData
import hu.bme.aut.tvshows.model.PersonSearchResult
import retrofit2.Call
import retrofit2.http.*

//retrofit2
interface PeopleApi {
    /**
     * Delete person
     *
     * @param personId  (required)
     * @return Call&lt;Void&gt;
     */
    @DELETE("people/{personId}")
    fun deletePeoplePersonId(
        @Path("personId") personId: Int
    ): Call<Void>

    /**
     * Get person by ID
     * Retrieve all primary information for a given person.
     * @param personId  (required)
     * @return Call&lt;Person&gt;
     */
    @GET("people/{personId}")
    fun getPeoplePersonId(
        @Path("personId") personId: Int
    ): Call<Person>

    /**
     * Search people
     *
     * @param q search text (optional)
     * @return Call&lt;List&lt;PersonSearchResult&gt;&gt;
     */
    @GET("search/people")
    fun getSearchPeople(
        @Query("q") q: String
    ): Call<List<PersonSearchResult>>

    /**
     * Create person
     *
     * @param body  (optional)
     * @return Call&lt;Person&gt;
     */
    @Headers("Content-Type:application/json")
    @POST("people")
    fun postPeople(
        @Body body: PersonData
    ): Call<Person>

    /**
     * Update person
     *
     * @param personId  (required)
     * @param body  (optional)
     * @return Call&lt;Void&gt;
     */
    @Headers("Content-Type:application/json")
    @PUT("people/{personId}")
    fun putPeoplePersonId(
        @Path("personId") personId: Int, @Body body: PersonData
    ): Call<Void>
}